### DISCUSSION TITLE
How to add class regex to tailwind LSP?

### DISCUSSION THREAD

**Discussion by SymphonySimper** [LazyVim/LazyVim#3054] [Q&A] [answered]
Here is what I'm trying to accomplish. [Example config from Class Variance Authority](https://cva.style/docs/getting-started/installation#intellisense)

Here is what I tried. But this does not seem to work.
```lua
return {
  "neovim/nvim-lspconfig",
  opts = {
    servers = {
      tailwindcss = {
        settings = {
          experimental = {
            classRegex = {
              { "cva\\(([^)]*)\\)", "[\"'`]([^\"'`]*).*?[\"'`]" },
              { "cx\\(([^)]*)\\)", "(?:'|\"|`)([^']*)(?:'|\"|`)" },
              { "cn\\(([^)]*)\\)", "[\"'`]([^\"'`]*).*?[\"'`]" },
              { "([a-zA-Z0-9\\-:]+)" },
            },
          },
        },
      },
    },
  },
}
```

**Comment by dpetka2001** [comment, marked as answer]
I believe it should be 
```lua
return {
  "neovim/nvim-lspconfig",
  opts = {
    servers = {
      tailwindcss = {
        settings = {
		  tailwindCSS = {
          experimental = {
            classRegex = {
              { "cva\\(([^)]*)\\)", "[\"'`]([^\"'`]*).*?[\"'`]" },
              { "cx\\(([^)]*)\\)", "(?:'|\"|`)([^']*)(?:'|\"|`)" },
              { "cn\\(([^)]*)\\)", "[\"'`]([^\"'`]*).*?[\"'`]" },
              { "([a-zA-Z0-9\\-:]+)" },
            },
          },
        },
      },
	},
    },
  },
}
```
I don't do tailwindcss, but this is what I gather from [lspconfig tailwindcss docs](https://github.com/neovim/nvim-lspconfig/blob/master/doc/server_configurations.md#tailwindcss).

**Reply by SymphonySimper** [reply]
That works. Thanks a lot!

**Reply by VitaliyR** [reply]
It doesn't work for me in the latest lazyvim, has anything changed in the config structure?

**Reply by maiermah** [reply]
does not work for me either

**Reply by DominickVale** [reply]
Same. Why isn't it working? Can we please get some help on this? Can we please unselect the answer since it's not working?  Otherwise nobody will try to help since they think it's solved.

**Reply by SymphonySimper** [reply]
![image](https://github.com/user-attachments/assets/8e9963a9-7716-46ae-a6dc-3d874810026a)
I am not using this custom regex anymore. But, just now tested with the marked solution and it seems to be working fine.

**Reply by x0aa7i** [reply]
Try it without tailwindcss extra, I think it overwrites the settings.

**Reply by SymphonySimper** [reply]
> Try it without tailwindcss extra, I think it overwrites the settings.

I'm not using LazyVim anymore. So not sure.

**Reply by Armadillidiid** [reply]
> Try it without tailwindcss extra, I think it overwrites the settings.

Disabling tailwindcss extra works. I tried inlining the regex patterns in the setup function, but that didn't work. It would be great if this issue could be resolved in the `lazyvim.plugins.extras.lang.tailwind` package for others in the future.

Sorry for tagging you, @folke
